cdap-security/src/main/java/io/cdap/cdap/security/tools/TokenGeneratorTool.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.tools;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.guice.InMemoryDiscoveryModule;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.security.auth.AbstractKeyManager;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.FileBasedKeyManager;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+
+/**
+ * A command line tool for generating access tokens.
+ */
+public class TokenGeneratorTool {
+
+  private void doMain(String[] args) throws ParseException, IOException {
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
+      .addOption(new Option("u", "username", true, "Username/principal"));
+
+    CommandLineParser parser = new BasicParser();
+    CommandLine commandLine = parser.parse(options, args);
+
+    if (commandLine.hasOption("h") || commandLine.getOptions().length == 0) {
+      printUsage(options);
+      return;
+    }
+
+    if (commandLine.hasOption("u")) {
+      String username = commandLine.getOptionValue("u");
+      printAccessToken(username);
+      return;
+    }
+
+    printUsage(options);
+  }
+
+  @VisibleForTesting
+  void printAccessToken(String username) throws IOException {
+    // Suppress logging from CDAP classes
+    ((Logger) LoggerFactory.getLogger(TokenManager.class)).setLevel(Level.ERROR);
```

Comment:
Any particular reason we want to supress the logging? Do we anticipate a lot of spam?

Reply:
This tool should just print the token to standard out. The output should be usable without having to filter out irrelevant log lines.
Without the logging suppression, we get a couple of extra log lines:
```
2022-09-20 14:10:00,926 - INFO  [main:i.c.c.s.a.TokenManager@46] - Starting TokenManager service
2022-09-20 14:10:01,198 - DEBUG [FileBasedKeyManager STARTING:i.c.c.s.a.FileBasedKeyManager@93] - Key 127113949 read from file data/security/keyfile
```